### DISCUSSION TITLE
Customizing the UI components

### DISCUSSION THREAD

**Discussion by smohith1803** [coder/code-server#6993] [Q&A]
Hi there,

I'm new to software development and I'm interested in learning how to customize the UI components of code-server. Could you please point me in the right direction or provide some resources that would be helpful?

i have gone through the code-server's codebase and couldn't find the source code of VScode , in fact the directory lib/vscode is empty. 
it would be very helpful, if you could point me to the part of the code-base where i could edit the ui components.

Thank you for your time and assistance.

**Comment by code-asher** [comment]
`lib/vscode` is a sub-module, so you need to run `git submodule update --init` to pull it down.  If you follow the contribution guide it will tell you more.  

But, if you are just experimenting it might be easier to clone VS Code directly (https://github.com/microsoft/vscode) and make your edits there rather than doing it through code-server.  code-server wraps VS Code with a web server and applies patches on top with `quilt` so it is a bit more complicated.

**Comment by smohith1803** [comment]
Thank You!! 

As part of my project, i need to make certain edits in the ui components of  code-server and build a docker image of it.

**Comment by bavjackson** [comment]
I'm also looking at doing something similar, seems like the way to go is to fork the current version of vscode, fork this repo, point `lib/vscode` at the forked vscode and then then build from that?

**Reply by code-asher** [reply]
If you plan on forking this repo and using the patch-based method, then you will not need to fork VS Code, you can just fork this repo, add your patches, then build.

But if you would rather not use patches, then yup you will need to fork both VS Code and code-server and point your code-server fork to your VS Code fork.

**Reply by bavjackson** [reply]
I'm not super familiar with writing patches so I'll spend a little time figuring it out and making a decision. I'm going to be doing a reasonable amount of UI work and replacing some of the default action bar items so it might be easier to go directly into vscode and then let code server just patch in the server/browser parts.

**Reply by code-asher** [reply]
Yeah if you have large changes planned, the patch method could be rough, so that makes sense.
